tests/Type/CollectionTest.php

```diff
@@ -265,7 +280,31 @@ public function removeFoo($value)
 		\Consistence\Type\Type::checkType($value, \'int\');
 		return \Consistence\Type\ArrayType\ArrayType::removeValue($this->children, $value);
 	}';
-		Assert::assertSame($method, $collection->generateMethod($propertyMetadata, $removeMethod));
+
+			return [
+				'propertyMetadata' => $propertyMetadata,
+				'sentryMethod' => $sentryMethod,
+				'expectedMethod' => $expectedMethod,
+			];
+		})();
+	}
+
+	/**
+	 * @dataProvider generateMethodDataProvider
+	 *
+	 * @param \Consistence\Sentry\Metadata\PropertyMetadata $propertyMetadata
+	 * @param \Consistence\Sentry\Metadata\SentryMethod $sentryMethod
+	 * @param string $expectedMethod
+	 */
+	public function testGenerateMethod(
+		PropertyMetadata $propertyMetadata,
+		SentryMethod $sentryMethod,
+		string $expectedMethod
```

Comment:
I think `expectedGeneratedMethod` would be more descriptive?

Reply:
Renamed.